docs/content/docs/concepts/Breaking Changes and Supported Patterns.md

```diff
@@ -0,0 +1,293 @@
+Unfortunately doesn't solve the problem: nearly everything is still incompatible change since TypeScript lets you traverse modules objects, inspect the code implementing functions etc.
+It's possible to write programs which break for pretty much any change.
+
+A more reasonable but simple supported pattern would be all code that only uses assumes the APIs behave as they are documented.
+This is more on the right track, but it has issues with ambiguity which tends to lead to one of:
+
+1. User and Package Developer not interpreting the documentation the same, and thus disagreeing on which changes are compatible.
+2. A broad interpretation of what behavior is supported, resulting in most changing being Incompatible.
+3. A very narrow interpretation of what behavior is supported, making authoring supported user code which actually benefits from using the package effectively impossible.
+4. Requiring too much documentation about explicitly what is and is not a supported use on every type.
+
+The rest of this document is about how we can take option 4 here, but structure that documentation in a centralized library of useful patterns which types can refer to to solve the verbosity problem.
+
+# Proposed Rules
+
+If this proposal is accepted, documents defining the rules in a friendly way for users to consume should be produced.
+The below can be considered an initial draft for what some of the rules might look like.
+
+## Version Type Variance
+
+### An Example: Modifying interfaces
+
+```typescript
+interface Name {
+  first: string;
+  last?: string;
+}
+```
+
+Here TypeScript is doing a lot for us.
+It defines what values are valid as a `Named`, and what types can be used as a `Named`.
+This isn't enough in practice to define a useful Supported Pattern for users of this interface.
+Consider the two following examples of User Code for it:
+
+```typescript
+// User Code 1
+const myName1: Name = { first: "My" };
+// User Code 2
+const myName2: Name = { first: "My", last: "Name" };
+// User Code 3
+interface FullName extends Name {
+  // Some people have multiple middle names, or none.
+  middle: string[];
+}
+```
+
+What ways can the package change `Named` without breaking any of these users?
+
+```typescript
+interface Name {
+  first: string;
+  last: string; // Make required. Causes User Code 1 to not build.
+}
+```
+
+```typescript
+interface Name {
+  first: string;
+  middle?: string; // Added optional field. Causes User Code 3 to not build.
+  last: string;
+}
+```
+
+```typescript
+interface Name {
+  first: string;
+  // Removed `last`. Causes User Code 2 to not build.
+}
+```
+
+### The Rule
+
+As shown in the example above, even when only consider if the code builds, its impossible to add or remove fields, optional or otherwise, or adjust the optionality of any field on an interface as a backwards compatible change if all three of these patterns are supported.
```

Comment:
```suggestion
As shown in the example above, even when only consider if the code builds, it's impossible to add or remove fields, optional or otherwise, or adjust the optionality of any field on an interface as a backwards compatible change if all three of these patterns are supported.
```